src/main/java/fr/openent/lystore/service/impl/DefaultPurseService.java

```diff
@@ -90,12 +107,34 @@ public void getPursesByCampaignId(Integer campaignId, Handler<Either<String, Jso
                 "left join orders " +
                 "               ON orders.id_structure = purse.id_structure AND  orders.id_campaign = purse.id_campaign " +
                 "WHERE  " +
-                "        purse.id_campaign = ? ;  " ;
+                "        purse.id_campaign = ? ;  ";
 
         JsonArray params = new fr.wseduc.webutils.collections.JsonArray()
                 .add(campaignId);
 
-        Sql.getInstance().prepared(query, params, SqlResult.validResultHandler(handler));
+        Sql.getInstance().prepared(query, params, SqlResult.validResultHandler(event -> {
+            if (event.isRight()) {
+                List<Purse> purses = new ArrayList<>();
+                JsonArray pursesJA = event.right().getValue();
+                for (int i = 0; i < pursesJA.size(); i++) {
+                    Purse purse = new Purse(pursesJA.getJsonObject(i));
+                    purses.add(purse);
+                }
```

Comment:
Tu peux utiliser `stream` pour avoir une liste de `Purse` pour chaque contenu

```suggestion
   List<Purse> purses = event.right().getValue().stream()
                        .filter(JsonObject.class::isInstance)
                        .map(JsonObject.class::cast)
                        .map(Purse::new)
                        .collect(Collectors.toList());
```
Tu filtre et map de sorte à ce que tu aies un JsonObject et tu remap afin de passer en référence ton objet

C'est l'équivalent de : 

```
 event.right().getValue().stream().map(purse -> new Purse((JsonObject) purse)).collect(Collectors.toList());
```

Dans ce cas tu "cast" ton objet mais du coup tu risques d'avoir une exception (warning dans un premier côté java)

Dans ma suggestion tu vérifies en sécurité ton objet avant de mapper en `Purse`